homeassistant/components/fritz/common.py

```diff
@@ -0,0 +1,235 @@
+"""Support for AVM FRITZ!Box classes."""
+from dataclasses import dataclass
+from datetime import datetime, timedelta
+import logging
+from typing import Any, Dict, Optional
+
+# pylint: disable=import-error
+from fritzconnection import FritzConnection
+from fritzconnection.lib.fritzhosts import FritzHosts
+from fritzconnection.lib.fritzstatus import FritzStatus
+
+from homeassistant.core import callback
+from homeassistant.helpers.dispatcher import async_dispatcher_send
+from homeassistant.helpers.event import async_track_time_interval
+from homeassistant.util import dt as dt_util
+
+from .const import (
+    DEFAULT_HOST,
+    DEFAULT_PORT,
+    DEFAULT_USERNAME,
+    DOMAIN,
+    TRACKER_SCAN_INTERVAL,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@dataclass
+class Device:
+    """FRITZ!Box device class."""
+
+    mac: str
+    ip_address: str
+    name: str
+
+
+class FritzBoxTools:
+    """FrtizBoxTools class."""
+
+    def __init__(
+        self,
+        hass,
+        password,
+        username=DEFAULT_USERNAME,
+        host=DEFAULT_HOST,
+        port=DEFAULT_PORT,
+    ):
+        """Initialize FritzboxTools class."""
+        self._cancel_scan = None
+        self._device_info = None
+        self._devices: Dict[str, Any] = {}
+        self._unique_id = None
+        self.connection = None
+        self.fritzhosts = None
+        self.fritzstatus = None
```

Comment:
We separate words with underscore in variable names. `fritz_status`

Reply:
@MartinHjelmare, will do.
Let me first complete the remaining platforms, PR 49699 and one more, and then I come back to this.

Simone